source/locationHelper.py

```diff
@@ -183,6 +178,65 @@ def toScreen(self, hwnd):
 class _RectMixin:
 	"""Mix-in class for properties shared between RectLTRB and RectLTWH classes"""
 
+	@classmethod
+	def fromNonInts(cls, *nonInts):
+		"""Creates an instance from parameters that aren't integers, such as floats.
+		The provided parameters will be converted to ints automatically.
+		@raise ValueError: If one of the input parameters can't be converted to int.
+		"""
+		return cls(*map(int, nonInts))
+
+	@classmethod
+	def fromCompatibleType(cls, rect):
+		"""Creates an instance from a compatible type.
+		Compatible types are defined in L{RECT_CLASSES}.
+		"""
+		if isinstance(rect,cls):
+			return rect
+		if isinstance(rect,RECT_CLASSES):
+			if cls is RectLTWH:
+				return cls(rect.left, rect.top, rect.right-rect.left, rect.bottom-rect.top)
+			return cls(rect.left, rect.top, rect.right, rect.bottom)

```

Comment:
If another type of `rect` class is introduced this could lead to subtle errors. Instead, do `elif cls is RectLTRB:` which will allow falling back to `raise TypeError`